src/LibDeclarations/webxr.nativeextensions.d.ts

```diff
@@ -1,11 +1,82 @@
-// This file contains native only extensions for WebXR  These APIs are not supported in the browser yet.
+// This file contains native only extensions for WebXR. These APIs are not supported in the browser yet.
 // They are intended for use with either Babylon Native https://github.com/BabylonJS/BabylonNative or
 // Babylon React Native: https://github.com/BabylonJS/BabylonReactNative
 
+// Open question: Do better types exist to use for vectors/quaternions?
+interface XRVector {
+    x: number;
+    y: number;
+    z: number;
+}
+
+interface XRQuaternion {
+    x: number;
+    y: number;
+    z: number;
+    w: number;
+}
+
+// Open question: Should XRGeometryType be defined or should all types be strings?
+type XRGeometryType = "unknown" | "background" | "wall" | "floor" | "ceiling" | "platform";
+
+interface XRFieldOfView {
+    angleLeft: number;
+    angleRight: number;
+    angleUp: number;
+    angleDown: number;
+}
+
+interface XRFrustum {
+    position: XRVector;
+    rotation: XRQuaternion;
+    fieldOfView: XRFieldOfView;
+    farDistance: number;
+}
+
+interface XRPlane {
+    // Open question: Should geometry id's and types be declared on XRPlanes or queried through other means?
+    geometryId?: number;
+    geometryType?: XRGeometryType | string;
+}
+
+// Open question: Should XRMesh have a concept of a XRReferenceSpace similar to XRPlane?
+interface XRMesh {

```

Comment:
Is it a mesh, or is it a geometry? I know we spoke semantics, but for me this is a geometry, that can be used to construct a mesh. The same way XRPlane is Plane the concept and not Plane the mesh-type.

Reply:
The term "geometry" may be ambiguous here.  This parent object is what we are calling `SceneObject`s in our native OpenXR extension.  I think of a scene object as being made up of some number of planes and meshes, with each of those being kinds of geometry.

Reply:
not opposed to renaming this as scene object. @RaananW does scene object or parentSceneObjectId/parentSceneObjectType make more sense here on your end?

Reply:
As you say, this will have to be documented correctly, but I have no problem with SceneObject and both parentGeometry variables

Reply:
Lets call these sceneobjects for now. I'll work to change these to be a class instance for comparison compared to relying on ids